### PR TITLE
Wallet removing refactor 

### PR DESCRIPTION
**Pull request by banklesss**

### CODE REVIEW

File: src/Catalyst/Step4.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ export const Step4 = ({pin, setVotingRegTxData}: Props) => {
       try {
         await ensureKeysValidity(wallet.id)
         navigation.navigate('biometrics', {
-          keyId: walletManager._id,
+          keyId: walletManager.id,
```
Comment (stackchain) [inline src/Catalyst/Step4.tsx]:
```suggestion
          keyId: walled.id,
```

File: src/Catalyst/Step4.tsx [hunk 2]
```diff
@@ -96,7 +96,7 @@ export const Step4 = ({pin, setVotingRegTxData}: Props) => {
       return
     }
     try {
-      const decryptedKey = await KeyStore.getData(walletManager._id, 'MASTER_PASSWORD', '', password, intl)
+      const decryptedKey = await KeyStore.getData(walletManager.id, 'MASTER_PASSWORD', '', password, intl)
```
Comment (stackchain) [inline src/Catalyst/Step4.tsx]:
```suggestion
      const decryptedKey = await KeyStore.getData(wallet.id, 'MASTER_PASSWORD', '', password, intl)
```

File: src/components/ConfirmTx/ConfirmTx.tsx [hunk 3]
```diff
@@ -158,7 +158,7 @@ export const ConfirmTx: React.FC<Props> = ({
             setDialogStep(DialogStep.WaitingHwResponse)
             signedTx = await wallet.signTxWithLedger(yoroiUnsignedTx, useUSB)
           } else {
-            const decryptedKey = await KeyStore.getData(walletManager._id, 'MASTER_PASSWORD', '', password, intl)
+            const decryptedKey = await KeyStore.getData(walletManager.id, 'MASTER_PASSWORD', '', password, intl)
```
Comment (stackchain) [inline src/components/ConfirmTx/ConfirmTx.tsx]:
```suggestion
            const decryptedKey = await KeyStore.getData(wallet.id, 'MASTER_PASSWORD', '', password, intl)
```

File: src/yoroi-wallets/walletManager.ts [hunk 4]
```diff
@@ -522,9 +518,8 @@ class WalletManager {
     await this.closeWallet()
   }
 
-  async removeCurrentWallet() {
+  async removeCurrentWallet(walletId: YoroiWallet['id']) {
```
Comment (stackchain) [inline src/yoroi-wallets/walletManager.ts]: I've updated our docs, walletId: string -> YoroiWallet['id']
```suggestion
  async removeWallet(walletId: YoroiWallet['id']) {
```

**Review by stackchain — CHANGES_REQUESTED** [review]
What was the issue? I'm guessing if the `walletMeta` is in sync with the selected wallet.

**Comment by banklesss** [comment]
> What was the issue? I'm guessing if the `walletMeta` is in sync with the selected wallet.

The wallet manager `_id` property is a wallet `id` copy, and we are using both separately. I think this could be a problem.

I am checking if there may be any desynchronization with the `walletMeta` at some point but I have not found it yet. If I find something I open a new PR

File: src/Catalyst/Step4.tsx (hunk 1 above)
Comment (banklesss) [reply src/Catalyst/Step4.tsx]: Thanks!

**Comment by wolverineks** [comment]
should this be merging into rc3?

**Comment by stackchain** [comment]
> should this be merging into rc3?

I don't think so.

**Comment by banklesss** [comment]
Closing. Changes don't needed.

**Pull request closed without merging**